lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -159,6 +165,25 @@ instance Functor m => Functor (NetworkLayer m target) where
         { nextBlocks = fmap (fmap f) . nextBlocks nl
         }
 
+-- Fetching the account balance may sometimes fail, especially on epoch
+-- boundaries. Therefore, we add some retrying mecanism here to cope with
+-- that nicely. 'readNextWithdrawal' is typically called by one-time
+-- handlers like 'postTransactions' and not by workers polling things
+-- regularly, so retrying a few times is okay here.
+getAccountBalanceRetry
+    :: NetworkLayer IO target block
+    -> ChimericAccount
+    -> ExceptT ErrGetAccountBalance IO (Quantity "lovelace" Word64)
+getAccountBalanceRetry nl account = ExceptT $
+    retrying policy decision (const $ runExceptT $ getAccountBalance nl account)
```

Comment:
I assume here, that there will be a moment, where cardano-node will be able to discover itn rewards basing on chimeric account? so itn and shelley chimeric account will be married, the same format, etc ? and we can reuse the same local query infrastructure already present?

Reply:
Indeed. In principle, on the 3rd of August, rewards will be transferred to the Mainnet based on the same public key that was used on the ITN.